dl/Util.py

```diff
@@ -227,6 +228,93 @@ def def_token(tok):
             return ANON_TOKEN
 
 
+def parse_auth_token(token):
+    """Parses string argument token
+    Usage:
+        parse_auth_token(token)
+
+    Parameters
+    ----------
+    token : str
+        A string auth token
+        E.g.
+        "testuser.3666.3666.$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/"
+
+    Returns
+    -------
+    return: a regex Match object or None
+    """
+    r"""
+    Explanation of the Regular Expression used:
+    E.g. token "testuser.3666.3666.$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/%"
+    Regex deconstruction and explanation:
+    -------------------------------------
+    1.   ([^\/\s]+)       any string with no "/" or spaces
+    2.   \.               separated by a .
+    3.   (\d+)            followed by any number of digits, user id
+    4.   \.               separated by a .
+    5.   (\d+)            followed by any number of digits, group id
+    6.   \.               separated by a .
+    7.a) (?:\$1\$\S{22,}) (Non capturing group) A string that starts with 
+                          $1$ (that's how a md5 hash starts)
+                          and that is followed by any non space characters
+                          of 22 chars or longer
+    7.b) |                or
+    7.c) (?:\S+_access)   (Non capturing group) A string that ends in _access.
+                          This is a special case for special tokens such as:
+                          anonymous.0.0.anon_access
+                          dldemo.99999.99999.demo_access
+    """
+
+    return re.match(r'([^\/\s]+)\.(\d+)\.(\d+)\.((?:\$1\$\S{22,})|(?:\S+_access))', token)
```

Comment:
We may want to anchor to the start and end of string as well depending on how you feel about this matching:

```Some other data       testuser.3666.3666.$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/          and some here```

The parsing still works it is just a matter of if we want to consider that valid.

Adding `^` to the beginning and `$` would clear this up eg.
```
^([^\/\s]+)\.(\d+)\.(\d+)\.((?:\$1\$\S{22,})|(?:\S+_access))$
```

Reply:
Very good observation! The reason I didn't go for that is because in the code I see often that they strip the string from blanks before they did the split looking for dots so I'm not sure what's going on, just to be safe I decided to leave it floating.